atlasdb-impl-shared/src/test/resources/lockwatch-event-cache-output/getEventsForTransactionsReturnsSnapshotWithOldEvents/event-cache-2.json

```diff
@@ -0,0 +1,51 @@
+{
+  "eventLog" : {
+    "snapshot" : {
+      "watches" : [ {
+        "type" : "fullTable",
+        "qualifiedTableRef" : "table"
+      } ],
+      "locked" : [ {
+        "bytes" : "dGFibGUAAQ=="
+      }, {
+        "bytes" : "dGFibGUAAw=="
+      } ],
+      "snapshotVersion" : {
+        "id" : "470c855e-f77b-44df-b56a-14d3df085dbc",
+        "version" : 6
```

Comment:
So the fact that snapshot is at 6, but current version is 10, and you only have event 10 means, next snapshot computation would fail right? I think you should make sure your tests don't construct wrong situations. This also hints that we should probably put detection around this into the code?

Reply:
There is detection in the code, but it happens when events are pushed to the snapshotter - this is tested in both `nonContiguousEventsThrows` and `snapshotMissedEventThrows`. However, this only occurs when events are aged out and passed to the snapshotter. I think the key thing here is that I should just fix the version here as you are right, should not construct bad situations unless we want them.